module/programs/helper.js

```diff
@@ -976,6 +977,153 @@ module.exports = class ProgramsHelper {
     });
   } 
 
+  /**
+  * Program join.
+  * @method
+  * @name join
+  * @param {String} programId - Program Id.
+  * @param {Object} data - body data.
+  * @param {String} userId - Logged in user id.
+  * @param {String} userToken - User token.
+  * @param {String} [appName = ""] - App Name.
+  * @param {String} [appVersion = ""] - App Version.
+  * @returns {Object} - Details of the program join.
+  */
+
+  static join( programId, data, userId, userToken, appName = "", appVersion = "", internalAccessToken = "" ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        
+        //Using programId fetch program name. Also checking the program status in the query.
+        let programData = await this.programDocuments({
+          _id: programId,
+          status: constants.common.ACTIVE
+        },["name", "externalId"]);
+        
+        if ( !programData.length > 0 ) {
+          throw ({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.PROGRAM_NOT_FOUND
+          });
+        }
+        
+        let programUsersData = {};
+        // Fetch user profile information by calling sunbird's user read api.
+        // !Important check specific fields of userProfile.
+        let userProfile = await userService.profile(userToken, userId);
+        if (!userProfile.success || 
+            !userProfile.data ||
+            !userProfile.data.response ||
+            !userProfile.data.response.profileUserTypes ||
+            !userProfile.data.response.profileUserTypes.length > 0 ||
+            !userProfile.data.response.userLocations ||
+            !userProfile.data.response.userLocations.length > 0
+        ) {
+          throw ({
+            status: httpStatusCode.bad_request.status,
+            message: constants.apiResponses.PROGRAM_JOIN_FAILED
+          });      
+        } 
+        programUsersData = {
+          programId: programId,
+          userRoleInformation: data.userRoleInformation ? data.userRoleInformation : {},
+          userId: userId,
+          userProfile: userProfile.data.response
+        }
+        if( appName != "" ) {
+          programUsersData['appInformation.appName'] = appName;
+        }
+        if( appVersion != "" ) {
+          programUsersData['appInformation.appVersion'] = appVersion;
+        }
+
+        //For internal calls add consent using sunbird api
+        if(internalAccessToken !== ""){
+          let consent = {
+            "request": {
+              "consent": {
+                "status": constants.common.REVOKED,
+                "userId": userProfile.data.response.id,
+                "consumerId": userProfile.data.response.organisations.organisationId,
```

Comment:
@VISHNUDAS-tunerlabs I think this is suppose to be program author org id. Also if so, we should mandate that in program creation API for support team and also update docs

Reply:
@aks30 yes. updated program model, key is given as programAuthorOrgId, if need any change we can make it